flake.nix

```diff
@@ -148,41 +191,26 @@
             name = "penumbra-and-cometbft";
             paths = [ penumbra cometbft ];
           };
-          devShells.default = craneLib.devShell {
-            inherit LIBCLANG_PATH ROCKSDB_LIB_DIR;
-            inputsFrom = [ penumbra ];
-            packages = [
-              buf
-              cargo-hack
-              cargo-nextest
-              cargo-release
-              cargo-watch
-              glibcLocales # for postgres initdb locale support
-              cometbft
-              grafana
-              grpcurl
-              grpcui
-              just
-              mdbook
-              mdbook-katex
-              mdbook-mermaid
-              mdbook-linkcheck
-              nix-prefetch-scripts
-              postgresql
-              process-compose
-              prometheus
-              protobuf
-              rocksdb
-              rsync
-              sqlfluff
-              toml-cli
-            ];
-            shellHook = ''
-              export LIBCLANG_PATH=${LIBCLANG_PATH}
-              export RUST_SRC_PATH=${pkgs.rustPlatform.rustLibSrc} # Required for rust-analyzer
-              export ROCKSDB_LIB_DIR=${ROCKSDB_LIB_DIR}
-              export RUST_LOG="info,network_integration=debug,pclientd=debug,pcli=info,pd=info,penumbra=info"
-            '';
+          devShells = {
+            default = craneLib.devShell {
+              inherit LIBCLANG_PATH ROCKSDB_LIB_DIR;
+              inputsFrom = [ penumbra ];
+              packages = commonDevPackages;
+              shellHook = ''
+                ${commonShellHook}
+                echo "Using stable Rust from rust-toolchain.toml"
+              '';
+            };
+
+            nightly = nightlyCraneLib.devShell {
+              inherit LIBCLANG_PATH ROCKSDB_LIB_DIR;
+              inputsFrom = [ penumbra ];
```

Comment:
I'm a bit unsure about this, because I think this will pull in whatever inputs are necessary to build the penumbra package, including the stable toolchain. Probably fine for now though, to the extent we care about the bloat, we can just directly make a derivation for the docs in the future.

Reply:
The `inputsFrom` declaration on the nightly shell was absolutely required: the nightly build failed without it, complaining about openssl paths. It's possible that we could decouple the inputs further, but for now, I consider it necessary.

It might be useful to have a "lightweight" devshell that's only rust, without all the dev-env requirements like cometbft, postgresql, etc. But I don't see a need for that now, so let's punt.